src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -166,10 +171,79 @@ private boolean validateNucleusConfig(CompletableFuture<DeploymentResult> totall
                         .complete(new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_NO_STATE_CHANGE, e));
                 return false;
             }
+
+            Integer timeoutSec = deployment.getDeploymentDocumentObj().getConfigurationValidationPolicy().timeoutInSeconds();
+            long configTimeout = Duration.ofSeconds(20).toMillis();
+            if (timeoutSec != null) {
+                configTimeout = Duration.ofSeconds(timeoutSec).toMillis();
+            }
+
+            Map<String, Object> currentConfig = kernel.getConfig().toPOJO();
+            if (configTimeout == 0
+                    || !deviceConfiguration.isDeviceConfiguredToTalkToCloud()
+                    || !hasNucleusConfigurationChange(currentConfig, nucleusConfig)) {
+                logger.atInfo().log("Skipping connectivity validation");
+                return true;
+            }
+            try {
+                // Update Nucleus with new config
+                logger.atInfo().log("Applying Nucleus config");
+                Map<String, Object> modifiedConfig = kernel.getConfig().toPOJO();
```

Comment:
reuse `currentConfig`?

Reply:
`currentConfig` is meant to be used when rolling back the changes.
I will rename it to `rollbackConfigs` so that's more clear.